Framework/Algorithms/test/ResampleXTest.h

```diff
@@ -171,158 +174,159 @@ class ResampleXTest : public CxxTest::TestSuite {
-    do_testResampleXEventWorkspace(WEIGHTED_NOTIME, true, true, true);
+    do_testResampleXEventWorkspace(WEIGHTED_NOTIME, true, true);
   }
 
   void testEventWorkspace_InPlace_NoPreserveEvents() {
-    do_testResampleXEventWorkspace(TOF, true, false, false);
+    do_testResampleXEventWorkspace(TOF, true, false);
   }
 
   void testEventWorkspace_InPlace_NoPreserveEvents_weighted() {
-    do_testResampleXEventWorkspace(WEIGHTED, true, false, false);
+    do_testResampleXEventWorkspace(WEIGHTED, true, false);
   }
 
   void testEventWorkspace_InPlace_NoPreserveEvents_weightedNoTime() {
-    do_testResampleXEventWorkspace(WEIGHTED_NOTIME, true, false, false);
+    do_testResampleXEventWorkspace(WEIGHTED_NOTIME, true, false);
   }
 
   void testEventWorkspace_NotInPlace_NoPreserveEvents() {
-    do_testResampleXEventWorkspace(TOF, false, false, false);
+    do_testResampleXEventWorkspace(TOF, false, false);
   }
 
   void testEventWorkspace_NotInPlace_NoPreserveEvents_weighted() {
-    do_testResampleXEventWorkspace(WEIGHTED, false, false, false);
+    do_testResampleXEventWorkspace(WEIGHTED, false, false);
   }
 
   void testEventWorkspace_NotInPlace_NoPreserveEvents_weightedNoTime() {
-    do_testResampleXEventWorkspace(WEIGHTED_NOTIME, false, false, false);
+    do_testResampleXEventWorkspace(WEIGHTED_NOTIME, false, false);
   }
 
   void testEventWorkspace_NotInPlace_PreserveEvents() {
-    do_testResampleXEventWorkspace(TOF, false, true, true);
+    do_testResampleXEventWorkspace(TOF, false, true);
   }
 
   void testEventWorkspace_NotInPlace_PreserveEvents_weighted() {
-    do_testResampleXEventWorkspace(WEIGHTED, false, true, true);
+    do_testResampleXEventWorkspace(WEIGHTED, false, true);
   }
 
   void testEventWorkspace_NotInPlace_PreserveEvents_weightedNoTime() {
-    do_testResampleXEventWorkspace(WEIGHTED_NOTIME, false, true, true);
+    do_testResampleXEventWorkspace(WEIGHTED_NOTIME, false, true);
   }
 
-  // Set-up a generic function for running tests for Workspace2D
+  // Set-up a generic function for running tests with Workspace2D
   void do_testResampleXWorkspace2D(bool inPlace, bool withDistribution) {
 
     std::string inName("test_inEvent");
     std::string outName("test_inEvent_output");
     if (inPlace)
       outName = inName;
 
-    Workspace2D_sptr test_in2D = Create2DWorkspace(100, 2);
-    test_in2D->setDistribution(withDistribution);
-    AnalysisDataService::Instance().add(inName, test_in2D);
+    int xlen = 100;
+    int ylen = 2;
+    double deltax = 0.75;
+    double countVal = 3.0;
+
+    Workspace2D_sptr ws = create<Workspace2D>(
+        ylen, Histogram(BinEdges(xlen + 1,
+                                 HistogramData::LinearGenerator(0.5, deltax)),
+                        Counts(xlen, countVal)));
+    ws->setDistribution(withDistribution);
+    AnalysisDataService::Instance().add(inName, ws);
+
+    // Check the size of the input workspace
```

Comment:
This is a test of `create`, so it should not be in this file (tests for this exist already in `WorkspaceCreationTest.h`). In general, avoid testing setup code, it just adds noise and makes the test hard to read.